www/kadm08/cv04/registration.php

```diff
@@ -0,0 +1,92 @@
+<?php include __DIR__ . '/lib/usersDb.php'; ?>
+<?php
+
+$errors = [];
+
+function sendEmail($recipient, $subject, $message) {
+    $headers = [
+        'MIME-Version: 1.0',
+        'Content-type: text/html, charset=utf-8',
+        'From: app@dev.com',
+        'Reply-To: app@dev.com',
+        'X-Mailer: PHP/8.0',
+    ];
+    $msg = "
+        <h1>Registration confirmation</h1>
+        <p>$message</p>
+    ";
+    return mail($recipient, $subject, $msg, implode("\r\n", $headers));
+}
+
+function registerNewUser($name, $email, $password) {
+    if (emailExists($email)) {
+        return [
+            'success' => false,
+            'message' => "User already exists!",
+        ];
+    }
+
+    addUser([
+        'name' => $name,
+        'email' => $email,
+        'password' => $password,
+    ]);
+    return [
+        'success' => true,
+        'message' => null,
+    ];
+}
+
+if (!empty($_POST)) {
+    $name =  htmlspecialchars(trim($_POST['name']));
+    $email =  htmlspecialchars(trim($_POST['email']));
+    $password =  htmlspecialchars(trim($_POST['password']));
+    $confirmPassword =  htmlspecialchars(trim($_POST['confirm']));
+
+    if ($password !== $confirmPassword) {
+        array_push($errors, 'Passwords do not match');
+    }
+
+    if (!$name) {
+        array_push($errors, 'Please enter your name');
+    }
+
+    if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+        array_push($errors, 'Please use a valid email');
+    }
+
+    if (strlen($password) < 3) {
+        array_push($errors, 'Passwords too short');
+    }
+
+    if (empty($errors)) {
+        $registrationResult = registerNewUser($name, $email, $password);
+        if ($registrationResult['success']) {
+            sendEmail($email, 'Registration was a success', "Hello, thank you for your registration, ...");
+            header("Location: login.php?email=$email");
+        } else {
+            array_push($errors, $registrationResult['message']);
+        }
+    }
+}
+
+?>
+
+
+<?php include __DIR__ . '/includes/head.php'; ?>
+<?php include __DIR__ . '/includes/navigation.php'; ?>
+<main>
+    <h1>Registration</h1>
+    <?php foreach($errors as $message): ?>
+        <p><?php echo $message; ?></p>
+    <?php endforeach; ?>
+    <form method="POST" action="<?php echo $_SERVER['PHP_SELF']; ?>" novalidate>
+        <input placeholder="name" name="name"><br>
+        <input placeholder="email" name="email" type="email"><br>

```

Comment:
chtelo by to zapamatovat vyplnene udaje ve formulari

Reply:
Opraveno.